library/azure_rm_trafficmanager.py

```diff
@@ -0,0 +1,588 @@
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Hai Cao <t-haicao@microsoft.com>"
+
+'''
+
+EXAMPLES = '''
+    - name: Create a Traffic Manager Profile
+      azure_rm_trafficmanager:
+        name: tmtest
+        resource_group: tmt
+        location: global
+        profile_status: Enabled
+        traffic_routing_method: Priority
+        dns_config:
+          relative_name: tmtest
+          ttl: 60
+        monitor_config:
+          protocol: https
+          port: 80
+          path: '/'
+        endpoints:
+          - name: e1
+            type: Microsoft.network/TrafficManagerProfiles/ExternalEndpoints
+            endpoint_location: West US 2
+            endpoint_status: Enabled
+            priority: 2
+            target: 1.2.3.4
+            weight: 1
+        tags:
+          Environment: Test
+
+    - name: Delete a Traffic Manager Profile
+      azure_rm_trafficmanager:
+        state: absent
+        name: tmtest
+        resource_group: tmt
+'''
+RETURN = '''
+state:
+    description: Current state of the Traffic Manager Profile
+    returned: always
+    type: dict
+    example:
+        "dns_config": {
+            "fqdn": "tmtest.trafficmanager.net",
+            "relative_name": "tmtest",
+            "ttl": 60
+        }
+        "endpoints": [
+            {
+                "endpoint_location": "West US 2",
+                "endpoint_monitor_status": "Degraded",
+                "endpoint_status": "Enabled",
+                "geo_mapping": null,
+                "id":   "/subscriptions/XXXXXX...XXXXXXXXX/resourceGroups/tmt/providers/Microsoft.Network/trafficManagerProfiles/tmtest/externalEndpoints/e1",
+                "min_child_endpoints": null,
+                "name": "e1",
+                "priority": 2,
+                "target": "1.2.3.4",
+                "target_resource_id": null,
+                "type": "Microsoft.Network/trafficManagerProfiles/externalEndpoints",
+                "weight": 1
+            }
+        ]
+        "id": "/subscriptions/XXXXXX...XXXXXXXXX/resourceGroups/tmt/providers/Microsoft.Network/trafficManagerProfiles/tmtest"
+        "location": "global"
+        "monitor_config": {
+            "interval_in_seconds": 30,
+            "path": "/",
+            "port": 80,
+            "profile_monitor_status": "Degraded",
+            "protocol": "HTTPS",
+            "timeout_in_seconds": 10,
+            "tolerated_number_of_failures": 3
+        }
+        "name": "tmtest"
+        "profile_status": "Enabled"
+        "tags": {
+            "Environment": "Test"
+        }
+        "traffic_routing_method": "Priority"
+        "type": "Microsoft.Network/trafficManagerProfiles"
+'''
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.mgmt.trafficmanager.models import (
```

Comment:
to avoid api version issue, we import model dynamically with version. Please refer [this usage](https://github.com/ansible/ansible/blob/b954917761eba2ddcdc8cf9dc811d668f7728eff/lib/ansible/modules/cloud/azure/azure_rm_virtualnetwork.py#L293) to declare the model in `azure_rm_common`